src/sentry/api/endpoints/project_group_index.py

```diff
@@ -800,29 +800,30 @@ def put(self, request, project):
             }
 
         if result.get('isPublic'):
-            queryset.update(is_public=True)
             for group in group_list:
-                if group.is_public:
-                    continue
-                group.is_public = True
-                Activity.objects.create(
+                share, created = GroupShare.objects.get_or_create(
```

Comment:
it might be nice to proviate a way to rotate the token too

Reply:
When you unshare and re-share it'd generate a new one. What case would you want to rotate one and not go through that? That'd mean exposing an API attribute in addition to `isPublic` to signify rotating?

Reply:
I'll leave this up to @ckj since he's going to help clean up the UI around this and if we want a "regenerate" button or whatever, that's trivial to add. Right now, it just looks like garbage anyways.

Reply:
Done with a91cae3d8669bd479e75d44cfec183f13cbce7f9